query_execution/QueryExecutionMessages.proto

```diff
@@ -31,14 +32,17 @@ message CatalogRelationNewBlockMessage {
 
   // Used by PartitionAwareInsertDestination.
   optional uint64 partition_id = 3;
+  required uint64 query_id = 4;
 }
 
 message DataPipelineMessage {
   required uint64 operator_index = 1;
   required fixed64 block_id = 2;
   required int32 relation_id = 3;
+  required uint64 query_id = 4;
 }
 
 message WorkOrdersAvailableMessage {
   required uint64 operator_index = 1;
+  required uint64 query_id = 2;
```

Comment:
It is also related to how we order things. It is nature for a message that first states the `query_id`, and then some detailed info within this query, i.e., `operator_index`.